Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -554,6 +554,13 @@ public final class PackageBuilder {
             // Add the created target to the map or print no sources warning.
             if let createdTarget = target {
                 targets[createdTarget.name] = createdTarget
+
+                // Check for duplicate target dependencies by name
+                let combinedDependencyNames = (createdTarget.dependencies.map { $0.name } + createdTarget.productDependencies.map { $0.0 })
```

Comment:
We should move this check into createTarget method.

Reply:
Moved into `createTarget`